webapp/templates/jury/analysis/contest_overview.html.twig

```diff
@@ -399,27 +421,30 @@ const statMap = submission_stats.reduce((map, stat) => {
 }, {});
 
 submissions.forEach(submission => {
-  let submission_minute = Math.floor((submission.submittime - submission.starttime) / 60);
-  let stat = statMap[submission.result];
-  if (stat && submission_minute >= 0 && submission_minute < contest_duration_minutes) {
-    stat.values[submission_minute][1]++;
+  const submission_bucket = Math.floor((submission.submittime - contest_start_time) / seconds_per_bucket);
+  const stat = statMap[submission.result];
+  if (stat && submission_bucket >= 0 && submission_bucket < bucket_count) {
+    stat.values[submission_bucket][1]++;
   }
 });
 
-for (let minute = 0; minute <= contest_duration_minutes; minute++) {
-  let this_minute_submission_nums = 0;
+let max_submissions_per_bucket = 1
+for (let bucket = 0; bucket < bucket_count; bucket++) {
+  let sum = 0;
   submission_stats.forEach(stat => {
-    this_minute_submission_nums += stat.values[minute][1];
+    sum += stat.values[bucket][1];
   });
-  max_submissions_per_minute = Math.max(max_submissions_per_minute, this_minute_submission_nums);
+  max_submissions_per_bucket = Math.max(max_submissions_per_bucket, sum);
 }
 
-// Pick a nice round tickDelta and tickValues
-var tickDelta = 15;
-while (contest_duration_minutes / tickDelta > 15) {
-  tickDelta *= 2;
-}
-var tickValues = Array.from({ length: Math.ceil(contest_duration_minutes / tickDelta) + 1 }, (_, i) => i * tickDelta);
+// Pick a nice round tickDelta and tickValues based on the step size of units.
+// We want whole values in the unit, and the ticks MUST match a corresponding bucket otherwise the resulting
+// coordinate will be NaN.
+const convert_factor = seconds_per_bucket / unit.convert;
+const maxTicks = Math.min(bucket_count, contest_duration / unit.step, 31)
```

Comment:
Where does this 31 come from?

Reply:
The idea was to limit the max number of ticks shown to some reasonable number. Changed it to the minimum bucket count as it looks pleasant.